odoo/models.py

```diff
@@ -1044,7 +1044,7 @@ def flush(*, xml_id=None, model=None):
         for id, xid, record, info in converted:
             if xid:
                 xid = xid if '.' in xid else "%s.%s" % (current_module, xid)
-                batch_xml_ids.update(ModelData._generate_xmlids(xid, self))
+                batch_xml_ids.add(xid)
```

Comment:
I am afraid this introduces a bug.  The purpose of that line was to be able to handle the following use-case:
- create a child record with an implicit parent record;
- create another child record with the same parent record as above.

When the second line is processed, the external id of the parent is found in the batch to flush.  So we flush and resume the external id resolution.

Reply:
> The purpose of that line was to be able to handle the following use-case:

> * create a child record with an implicit parent record;
> * create another child record with the same parent record as above.

Is that actually a use-case we want to support? Because not only does it sounds really arbitrary (and somewhat stupid given the requirement that whoever craft their import has to know about the parent generation ruleset *and* refuse to pre-create their parent records), it also means this PR would break the use case if this feature is used with an explicit parent.

Reply:
This use-case is really tricky to do with CSV or any "tabular" file.  However it is very easy to do with `<record>` in XML files, which were intended to be processed in batch, but never done.  **So I am fine with the change.**
Note that most developers simply don't know about this magic XML id creation for parent records, so mostly nobody uses it.  That's why I am not against dropping completely the feature.